### PR TITLE
simplify Layers.test usage

### PR DESCRIPTION
**Pull request by felixmariotto**
Related to a thread on the forum [here](https://discourse.threejs.org/t/getting-visible-layers-of-camera/22036)

It is a bit counter-intuitive to use [Layers.test](https://threejs.org/docs/#api/en/core/Layers.test) at the moment, because it takes a `Layers` object, whereas all other methods of `Layers` taking an argument take an integer. You might want to test if a layer is visible without knowing what object is a member of it, or not having this object in scope, which make it necessary to use a dummy object.

**Comparison : testing if camera is a member of layer 2**

Before :
```javascript
const dummy = new THREE.Object3D();
dummy.layers.set( 2 );
camera.layers.test( dummy.layers );
```

After : 
```javascript
camera.layers.testIndex( 2 );
```

### CODE REVIEW

**Comment by Mugen87** [comment]
Instead of introducing a new method, we could also use the same approach like `lookAt()`. Meaning `test()` is defined like so:
```ts
test( layers: Layers | number ): boolean;
```
The implementation would change to:
```js
test( layers ) {

   let mask = ( layers.isLayers ) ? layers.mask : layer;

   return ( this.mask & mask ) !== 0;

}
```
It's only necessary to define `isLayers`.

I guess I vote for this approach 😇 .

**Comment by Mugen87** [comment]
Instead of introducing a new method, we could also use the same approach like `lookAt()`. Meaning `test()` is defined like so:
```ts
test( layers: Layers | number ): boolean;
```

**Comment by felixmariotto** [comment]
I would prefer having the option to use `Layers.test` too, but I was wary that adding a type check might impact performances a bit. `Layers.test` seems to be used every frame for every object [here](https://github.com/mrdoob/three.js/blob/670b1e9e85356d98efa4c702e93c85dd52f01e1e/src/renderers/WebGLRenderer.js#L1134). Would it be ok to have this ?

```javascript
test( layers ) {

	const mask = ( typeof layers === "number" ) ? ( 1 << layers | 0 ) : layers.mask;

	return ( this.mask & mask ) !== 0;

}
```

**Comment by Mugen87** [comment]
I guess we need a performance measurement to know the impact of the additional test for complex scene graphs.

**Comment by felixmariotto** [comment]
Here are two performance tests : 

**[Original Layers.test](https://jsfiddle.net/felixmariotto/8p7365c1/104/)**
scene with either 100, 1000 or 10000 times one point in a 10 levels hierarchy ( so either 1000, 10000 or 100000 object3Ds ).

**[Modified Layers.test](https://jsfiddle.net/felixmariotto/xv26rLnp/12/)**
same scene as before, but we update `Layers.prototype.test` to this : 
```javascript
test( layers ) {

	const mask = ( typeof layers === "number" ) ? ( 1 << layers | 0 ) : layers.mask;

	return ( this.mask & mask ) !== 0;

}
```
See the console for the result of the new method usage.

For both scenes I get a steady 20FPS on my laptop with 100000 object3Ds, so it seems not to impact performance, but let me know if we need more test.

**Comment by mrdoob** [comment]
I would rather add a new method instead.
Maybe make `test` use integers and add `testLayer` method or a new `testInteger` method?

File: docs/api/en/core/Layers.html [hunk 1]
```diff
@@ -72,6 +72,13 @@ <h3>[method:Boolean test]( [param:Layers layers] )</h3>
 			Returns true if this and the passed *layers* object have at least one layer in common.
 		</p>
 
+		<h3>[method:Boolean testInteger]( [param:Integer layer] )</h3>
+		<p>
+			layer - an integer from 0 to 31.<br /><br />
```
Comment (mrdoob) [inline docs/api/en/core/Layers.html:77]: 0 to 31?

Comment (felixmariotto) [reply docs/api/en/core/Layers.html:77]: It seems from the doc that 

> A Layers object assigns an Object3D to 1 or more of 32 layers numbered 0 to 31

Should I change something ?

Comment (Mugen87) [reply docs/api/en/core/Layers.html:77]: I think @felixmariotto was correct here. Valid channel are in the integer range `[0,31]`.

BTW: I would rename the method to `testChannel()` and add the discription: `Whether the given channel is enabled or not.

**Comment by Mugen87** [comment]
Closing in favor of #22825.

**Pull request closed without merging**